packages/apollo-engine-reporting/src/plugin.ts

```diff
@@ -138,7 +155,18 @@ export const plugin = <TContext>(
       let didResolveSource: boolean = false;
 
       return {
+        didResolveOperation(requestContext) {
+          if (typeof options.traceReporting === 'function') {
+            options.traceReporting(requestContext).then(shouldReportTrace => {
+              if (!shouldReportTrace) didEnd(requestContext, false);
+            });
+          }
+        },
```

Comment:
This `didResolveOperation` method seems like it should be `await`ed (and thus allow execution — and the subsequent `willResolveField`s to yield) in order to allow asynchronous behavior within the user-defined `traceReporting` predicate to take place (e.g., calling out to a network service).

```suggestion
        async didResolveOperation(requestContext) {
          if (typeof options.traceReporting !== 'function') return;
          const shouldReportTrace = await options.traceReporting(requestContext);
          if (!shouldReportTrace) didEnd(requestContext, false);
        },
```